File: python/n520_SearchAlp.py
# https://leetcode-cn.com/problems/detect-capital/
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        if word[0:1].isupper():
            if word[1:].islower() or word[1:].isupper() or len(word) == 1: return True
            return False
        if word.islower(): return True
        return False 
        

if __name__ == "__main__":
    pass       
            

File: cpp/n520_SearchAlp.cpp
// https://leetcode-cn.com/problems/detect-capital/
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    bool detectCapitalUse(string word) {
        if (word.size() == 1) return true;
        if (isupper(word[0])){
            int tmp = isupper(word[1]);
            if (word.size()==2) return true;
            for (int i = 2; i < word.size(); ++i)
                if (isupper(word[i]) != tmp) return false;
            return true;
        }
        else{
            for (int i = 1; i < word.size(); ++i)
                if (isupper(word[i])) return false;
            return true;
        }
    }
};

int main(){
    return 0;
}